module3/homework/inner_product/arithmeticAverage.cpp

```diff
@@ -0,0 +1,18 @@
+#include "arithmeticAverage.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <numeric>
+
+double ArithmeticAverage(std::vector<int> first, std::vector<int> second) {
+    return (double)std::inner_product(first.begin(), first.end(), second.begin(), 0, std::plus<>(), std::plus<>()) /
+           (first.size() + second.size());
+}
+
+double Distance(std::vector<int> first, std::vector<int> second) {
+    return sqrt(std::inner_product(first.begin(), first.end(), second.begin(), 0.0, std::plus<>(),
+                                   [](auto& el1, auto& el2) {
+                                       return pow(el1 - el2, 2);
+                                       }));
+}
```

Comment:
What do You think about using constant iterators? I think that You will not modify them in this exercise. Of course it isn't any issue. It's just good practice in my opinion.

Reply:
done